app/src/main/java/com/android/doctorapp/ui/profile/ProfileViewModel.kt

```diff
@@ -51,6 +73,45 @@ class ProfileViewModel @Inject constructor(
         }
     }
 
+    fun getUserProfileData() {
+        viewModelScope.launch {
+            var recordId: String = ""
+            session.getString(USER_ID).collectLatest {
+                recordId = it.orEmpty()
+                if (context.isNetworkAvailable()) {
+                    setShowProgress(true)
+                    when (val response =
+                        profileRepository.getProfileRecordById(recordId, fireStore)) {
+                        is ApiSuccessResponse -> {
+                            userProfileDataResponse.value = response.body!!
```

Comment:
use this    userProfileDataResponse object to show age. i think no need to extra age variable.